spec/unit/models/user_filters/user_filter_builder_spec.rb

```diff
@@ -0,0 +1,100 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+describe GoodData::UserFilterBuilder do
+  describe '.execute_mufs' do
+    let(:login) { 'rubydev+admin@gooddata.com' }
+    let(:full_definition_filter) do
+      {
+        :login=> login,
+        :filters=>[{
+          :label=>label_id,
+          :over=>nil,
+          :to=>nil,
+          :values=>["Washington"]}]
+      }
+    end
+
+    let(:label_id) { 'label.csv_policies.state' }
+    let(:label_uri) { '/gdc/md/wock3futg594lz3ornqv70yvbsivf896/obj/270' }
+    let(:filter_definitions) { [full_definition_filter] }
+    let(:options) { { client: client,
+                      project: project } }
+    let(:users_brick_input) { [{ 'login' => login }] }
+    let(:client) { double('client') }
+    let(:project) { double('project') }
+    let(:user) { double('user')}
+    let(:project_users) { [user] }
+    let(:label) { double('label') }
+    let(:filter) { double('filter') }
+    let(:existing_filter) { double('existing_filter') }
+
+    before do
+      allow(project).to receive(:labels).with(label_uri)
+        .and_return(label)
+      allow(project).to receive(:labels).with("label.csv_policies.state")
+        .and_return(label)
+      allow(project).to receive(:attributes)
+      allow(project).to receive(:users).and_return([])
+      allow(project).to receive(:data_permissions).and_return([existing_filter])
+      allow(project).to receive(:pid)
+      allow(label).to receive(:values_count).and_return(666_666)
+      allow(label).to receive(:uri).and_return(label_uri)
+      allow(label).to receive(:find_value_uri).with('Washington').and_return('foo')
+      allow(label).to receive(:attribute_uri).and_return('bar')
+      allow(client).to receive(:create).and_return(filter)
+      allow(client).to receive(:get)
+        .and_return({ 'userFilters' => { 'items' => [] } })
+      allow(client).to receive(:post)
+        .and_return({ 'userFiltersUpdateResult' => [] })
+      allow(filter).to receive(:related_uri)
+      allow(existing_filter).to receive(:related_uri)
+      allow(filter).to receive(:save)
+      allow(filter).to receive(:uri)
+    end
+
+    it 'resolves mufs to be created/deleted' do
```

Comment:
Similar code found in user_filter_builder_spec.rb:59, user_filter_builder_spec.rb:78 (mass = 60)